include/inexor/vulkan-renderer/imgui/imgui.hpp

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#include "inexor/vulkan-renderer/wrapper/command_buffer.hpp"
+#include "inexor/vulkan-renderer/wrapper/command_pool.hpp"
+#include "inexor/vulkan-renderer/wrapper/device.hpp"
+#include "inexor/vulkan-renderer/wrapper/framebuffer.hpp"
+#include "inexor/vulkan-renderer/wrapper/graphics_pipeline.hpp"
+#include "inexor/vulkan-renderer/wrapper/make_info.hpp"
+#include "inexor/vulkan-renderer/wrapper/mesh_buffer.hpp"
+#include "inexor/vulkan-renderer/wrapper/renderpass.hpp"
+#include "inexor/vulkan-renderer/wrapper/resource_descriptor.hpp"
+#include "inexor/vulkan-renderer/wrapper/shader.hpp"
+#include "inexor/vulkan-renderer/wrapper/swapchain.hpp"
+#include "inexor/vulkan-renderer/wrapper/texture.hpp"
+
+#include <glm/glm.hpp>
+#include <imgui.h>
+#include <spdlog/spdlog.h>
+#include <vma/vk_mem_alloc.h>
+#include <vulkan/vulkan.h>
+
+#include <assert.h>
+#include <iomanip>
+#include <memory>
+#include <sstream>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <vector>
+
+namespace inexor::vulkan_renderer::imgui {
+class ImGUIOverlay {
+    const wrapper::Device &m_device;
+    const wrapper::Swapchain &m_swapchain;
+    const VkPhysicalDevice m_graphics_card;
+    const VmaAllocator m_vma_allocator;
+    const VkQueue m_data_transfer_queue;
+    const std::uint32_t m_data_transfer_queue_family_index;
+
+    VkPipelineLayout m_pipeline_layout;
+    VkSampleCountFlagBits rasterizationSamples{VK_SAMPLE_COUNT_1_BIT};
```

Comment:
This can probably just be inlined.

Reply:
Right. The following member was already removed because it can be accessed through `wrapper::Device`: 

```cpp
const std::uint32_t m_data_transfer_queue_family_index;
```

Reply:
`VkPipelineLayout m_pipeline_layout;` however can't be inlined: It's used in `ImGUIOverlay::prepare_drawing()`.

Reply:
Done.